tasks/testwasher.py

```diff
@@ -67,6 +61,26 @@ def parse_flaky_file(self):
             for package, tests in flakes.items():
                 self.known_flaky_tests[f"github.com/DataDog/datadog-agent/{package}"].update(set(tests))
 
+    def parse_test_results(self, module_path: str) -> Tuple[dict, dict]:
+        failing_tests = defaultdict(set)
+        flaky_marked_tests = defaultdict(set)
+
+        with open(f"{module_path}/{self.test_output_json_file}", encoding='utf-8') as f:
+            for line in f:
+                test_result = json.loads(line)
+                if test_result["Action"] == "fail" and "Test" in test_result:
+                    failing_tests[test_result["Package"]].add(test_result["Test"])
+                if test_result["Action"] == "success" and "Test" in test_result:
+                    if test_result["Test"] in failing_tests[test_result["Package"]]:
+                        failing_tests[test_result["Package"]].remove(test_result["Test"])
+                if (
+                    "Output" in test_result
+                    and "Test" in test_result
+                    and self.flaky_test_indicator in test_result["Output"]
+                ):
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏`"Test" in test_result` is in all conditions, this can be cleaned